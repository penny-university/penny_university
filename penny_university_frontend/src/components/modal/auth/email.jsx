// @flow

import React, { useState } from 'react'
import { connect } from 'react-redux'
import {
  Form, ModalHeader, ModalBody, Button,
} from 'reactstrap'
import propTypes from 'prop-types'
import { Input } from '../../fields'
import modalDispatch from '../dispatch'
import { dispatchUserExists } from '../../../actions/user'

const AuthEmailModal = ({ userExists }) => {
  const [email, setEmail] = useState('')
  return (
    <>
      <ModalHeader toggle={modalDispatch.close}>Sign up or log in</ModalHeader>
      <ModalBody>
        <Form onSubmit={(e) => {
          e.preventDefault()
          userExists(email).then((action) => {
            if (action.payload?.status === 400) {
              modalDispatch.authSignup(email)
            } else {
              modalDispatch.authPassword(email)
            }
          })
          return false
        }}
        >
          <Input label="Email" type="email" name="email" id="email" placeholder="" required onChange={setEmail} value={email} />
          <div className="text-center">
            <Button>
              Let&rsquo;s Go
            </Button>
          </div>
        </Form>
      </ModalBody>
    </>
  )
}


const mapStateToProps = () => ({
})

const mapDispatchToProps = (dispatch) => ({
  userExists: (email) => dispatch(dispatchUserExists(email)),
})
AuthEmailModal.propTypes = {
  userExists: propTypes.bool.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(AuthEmailModal)
