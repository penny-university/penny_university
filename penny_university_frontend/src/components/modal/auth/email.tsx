// @flow

import React, { useState } from 'react'
import { connect } from 'react-redux'
import { ThunkDispatch } from 'redux-thunk'
import { AnyAction } from 'redux'
import {
  Form, ModalHeader, ModalBody, Button,
} from 'reactstrap'
import { Input } from '../../fields'
import modalDispatch from '../dispatch'
import { userExists } from '../../../actions/user'

type AuthEmailModalProps = {
  userExists: (email: string, followUp: { chatId: number, content: string } | undefined) => AnyAction
  followUp?: { chatId: number, content: string } | undefined,
}

const AuthEmailModal = ({ userExists, followUp }: AuthEmailModalProps) => {
  const [email, setEmail] = useState('')
  return (
    <>
      <ModalHeader toggle={modalDispatch.close}>Sign up or log in</ModalHeader>
      <ModalBody>
        <Form onSubmit={(e) => {
          e.preventDefault()
          userExists(email, followUp)
          return false
        }}
        >
          <Input label="Email" type="email" name="email" id="email" placeholder="" required onChange={setEmail} value={email} />
          <div className="text-center">
            <Button>
              Let&rsquo;s Go
            </Button>
          </div>
        </Form>
      </ModalBody>
    </>
  )
}

AuthEmailModal.defaultProps = {
  followUp: undefined,
}


const mapStateToProps = () => ({
})

const mapDispatchToProps = (dispatch: ThunkDispatch<{}, {}, AnyAction>) => ({
  userExists: (email: string, followUp: { chatId: number, content: string } | undefined) => dispatch(userExists(email, followUp)),
})
export default connect(mapStateToProps, mapDispatchToProps)(AuthEmailModal)
