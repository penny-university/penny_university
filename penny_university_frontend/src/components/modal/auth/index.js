// @flow
import AuthEmail from './email'
import AuthPassword from './password'
import AuthSignup from './signup'


export {
  AuthEmail,
  AuthPassword,
  AuthSignup,
}
