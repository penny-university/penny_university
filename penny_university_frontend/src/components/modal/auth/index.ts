// @flow
import AuthEmail from './email'
import AuthPassword from './password'
import AuthPasswordReset from './reset';
import AuthSignup from './signup'
import VerifyEmail from './verify'

export {
  AuthEmail,
  AuthPassword,
  AuthPasswordReset,
  AuthSignup,
  VerifyEmail,
}
