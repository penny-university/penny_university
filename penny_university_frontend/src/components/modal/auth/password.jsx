// @flow

import React, { useState } from 'react'
import { connect } from 'react-redux'
import {
  Form, ModalHeader, ModalBody, Button,
} from 'reactstrap'
import propTypes from 'prop-types'
import modalDispatch from '../dispatch'
import { dispatchLogin } from '../../../actions/user'
import { Input } from '../../fields'

const AuthPasswordModal = ({ email, login }) => {
  const [password, setPassword] = useState('')
  return (
    <>
      <ModalHeader toggle={modalDispatch.close}>Welcome back!</ModalHeader>
      <ModalBody>
        <Form onSubmit={(e) => {
          e.preventDefault()
          login({ email, password })
        }}
        >
          <Input label="Password" type="password" name="password" id="password" placeholder="" onChange={setPassword} value={password} required />
          <div className="text-center">
            <Button>Let&rsquo;s Go</Button>
          </div>
        </Form>
      </ModalBody>
    </>
  )
}


const mapStateToProps = () => ({
})

const mapDispatchToProps = (dispatch) => ({
  login: (payload) => dispatch(dispatchLogin(payload)),
})

AuthPasswordModal.propTypes = {
  login: propTypes.bool.isRequired,
  email: propTypes.string.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(AuthPasswordModal)
