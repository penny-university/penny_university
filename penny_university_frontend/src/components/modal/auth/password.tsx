// @flow

import React, { useState } from 'react'
import { connect } from 'react-redux'
import { ThunkDispatch } from 'redux-thunk'
import { AnyAction } from 'redux'
import {
  Form, ModalHeader, ModalBody, Button,
} from 'reactstrap'
import modalDispatch from '../dispatch'
import { dispatchLogin } from '../../../actions/user'
import { Input } from '../../fields'

type AuthPasswordModalProps = {
  email: string,
  followUp?: { chatId: number, content: string }  | undefined,
  login: (payload: {email: string, password: string, followUp: { chatId: number, content: string } | undefined }) => void,
}

const AuthPasswordModal = ({ email, login, followUp }: AuthPasswordModalProps) => {
  const [password, setPassword] = useState('')
  return (
    <>
      <ModalHeader toggle={modalDispatch.close}>Welcome back!</ModalHeader>
      <ModalBody>
        <Form onSubmit={(e) => {
          e.preventDefault()
          login({ email, password, followUp })
        }}
        >
          <Input label="Password" type="password" name="password" id="password" placeholder="" onChange={setPassword} value={password} required />
          <div className="text-center">
            <Button>Let&rsquo;s Go</Button>
          </div>
        </Form>
      </ModalBody>
    </>
  )
}

AuthPasswordModal.defaultProps = {
  followUp: undefined,
}


const mapStateToProps = () => ({
})

const mapDispatchToProps = (dispatch: ThunkDispatch<{}, {}, AnyAction>) => ({
  login: (payload: {email: string, password: string, followUp: { chatId: number, content: string } | undefined }) => dispatch(dispatchLogin(payload)),
})


export default connect(mapStateToProps, mapDispatchToProps)(AuthPasswordModal)
