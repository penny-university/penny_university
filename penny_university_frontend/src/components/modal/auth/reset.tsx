import React from 'react'
import {
  ModalHeader, ModalBody,
} from 'reactstrap'
import modalDispatch from '../dispatch'

type AuthPasswordResetModalProps = {
  email: string
}

const AuthPasswordResetModal = ({ email }: AuthPasswordResetModalProps) => {
  return (
    <>
      <ModalHeader toggle={modalDispatch.close}>Reset Password</ModalHeader>
      <ModalBody>
        Password reset request sent to {email}!
      </ModalBody>
    </>
  )
}


export default AuthPasswordResetModal
