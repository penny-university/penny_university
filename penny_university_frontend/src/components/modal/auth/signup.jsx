// @flow

import React, { useState } from 'react'
import { connect } from 'react-redux'
import {
  Form, ModalHeader, ModalBody, Button,
} from 'reactstrap'
import propTypes from 'prop-types'
import modalDispatch from '../dispatch'
import { dispatchSignup } from '../../../actions/user'
import { Input } from '../../fields'


const AuthSignupModal = ({ signup, email: orignalEmail }) => {
  const [firstName, setFirstName] = useState('')
  const [lastName, setLastName] = useState('')
  const [password, setPassword] = useState('')
  const [email, setEmail] = useState(orignalEmail)
  return (
    <>
      <ModalHeader toggle={modalDispatch.close}>Looks like you're new. Fill out this form to create an account.</ModalHeader>
      <ModalBody>
        <Form onSubmit={(e) => {
          e.preventDefault()
          signup({
            password, firstName, lastName, email,
          })
        }}
        >
          <Input label="First Name" type="text" name="firstName" id="firstName" placeholder="" onChange={setFirstName} value={firstName} required />
          <Input label="Last Name" type="text" name="lastName" id="lastName" placeholder="" onChange={setLastName} value={lastName} required />
          <Input label="Email" type="email" name="email" id="email" placeholder="" onChange={setEmail} value={email} helperText={"If you are part of the Penny U Slack, use the same email as that slack account."} />
          <Input label="Password" type="password" name="password" id="password" placeholder="" onChange={setPassword} value={password} required />
          <div className="text-center">
            <Button>
              Let&rsquo;s Go
            </Button>
          </div>
        </Form>
      </ModalBody>
    </>
  )
}


const mapStateToProps = () => ({
})

const mapDispatchToProps = (dispatch) => ({
  signup: (payload) => dispatch(dispatchSignup(payload)),
})

AuthSignupModal.propTypes = {
  signup: propTypes.bool.isRequired,
  email: propTypes.string.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(AuthSignupModal)
