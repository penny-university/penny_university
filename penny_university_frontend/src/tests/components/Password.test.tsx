import React from 'react'
import { shallow, mount } from 'enzyme'
import { AuthPasswordModal } from '../../components/modal/auth/password'

describe('auth password modal', () => {
  it('renders correctly', () => {
    const email = 'test@gmail.com'
    const component = shallow(<AuthPasswordModal email={email} login={jest.fn()} requestPasswordReset={jest.fn()} />)
    expect(component).toMatchSnapshot()
  })

  it('should call requestPasswordReset upon clicking forgot password', () => {
    const email = 'test@gmail.com'
    const requestPasswordReset = jest.fn()
    const component = mount(
      <AuthPasswordModal email={email} login={jest.fn()} requestPasswordReset={requestPasswordReset} />,
    )
    const forgotPassword = component.find('#forgot-password').find('button')
    forgotPassword.simulate('click')
    expect(requestPasswordReset.mock.calls[0][0]).toEqual({ email: 'test@gmail.com' })
  })
})
