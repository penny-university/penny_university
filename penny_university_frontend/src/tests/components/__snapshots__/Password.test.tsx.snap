// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth password modal renders correctly 1`] = `
<Fragment>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    toggle={[Function]}
    wrapTag="div"
  >
    Welcome back!
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <Form
      onSubmit={[Function]}
      tag="form"
    >
      <Input
        helperText=""
        id="password"
        label="Password"
        name="password"
        onChange={[Function]}
        placeholder=""
        required={true}
        type="password"
        value=""
      />
      <div
        className="text-center"
      >
        <Button
          color="secondary"
          tag="button"
        >
          Letâ€™s Go
        </Button>
      </div>
    </Form>
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      color="link"
      id="forgot-password"
      onClick={[Function]}
      tag="button"
    >
      Forgot password?
    </Button>
  </ModalFooter>
</Fragment>
`;
